doc/index.md

```diff
@@ -64,37 +122,58 @@ Cqueues can be integrated with almost any main loop or event library you may enc
 
 All connection types expose the fields:
 
+
 ### `connection.type` <!-- --> {#connection.type}
 
-Either `"client"` or `"server"`
+The mode of use for the connection object. Valid values are:
+- `"client"` - Connects to a remote uri
+- `"server"` - Listens for connection on a local uri
 
 
 ### `connection.version` <!-- --> {#connection.version}
 
-The HTTP version as a number
+The HTTP version number of the connection. 
 
 
 ### `connection:connect(timeout)` <!-- --> {#connection:connect}
 
+Connects to a remote server using the address specified, HTTP version and any options specified in the `connection.new` consturctor. The `connect` function will yield until the connection attempt finishes (success or failure) or until `timeout` is exceeded. Connecting may include DNS lookups, TLS negotiation and HTTP2 settings exchange. Returns true on success. On failure returns nil and an error message.
```

Comment:
"Connects to a remote server" => "Completes the connection to the remote server"